### DISCUSSION TITLE
Nextjs google site verification with next-intl

### DISCUSSION THREAD

**Discussion by aboutjquery** [amannn/next-intl#1410] [Q&A]
I am using Nextjs + next-intl. I now need google site verification. I chose to add meta, but it was unsuccessful because `mydomain.com` will automatically jump to `/en/ or /zh/` route depending on the language because `app/page.tsx` has below code:

```
export default function RootPage() {
  redirect('/en');
}
```

I tried adding below code to `app/layout.tsx`

```
export const metadata: Metadata = {
  verification: {
    google: '...',
  },
};
```

And then clicked the Google search console verification button, but it was unsuccessful. I tried adding meta to `app/[locale]/layout.tsx` and it succeeded, but it can only be verified when the URL is the Language. , but I want to verify the URL without language routing, am I wrong? I don’t need this? I just need to add meta to the two locales respectively?

**Comment by amannn** [comment]
You might have an easier time by using the [HTML file upload](https://support.google.com/webmasters/answer/9008080?hl=en#html_verification&zippy=%2Chtml-file-upload) method.

**Reply by aboutjquery** [reply]
Thank you for your answer. I am using the Droplets of DxxxxxOxxxx to place my `nextjs + next-intl` project.

According to google instructions I should upload to the root directory, but I don't know where the root directory of the nextjs project is ;)

After I run `npm run build`, the `/.next/` directory was generated. I think that is the root directory ;)

I tried uploading the html file to

```
/var/www/mydomain/
/var/www/mydomain/.next/
/var/www/mydomain/.next/server/app/
/var/www/mydomain/public/
```

But all failed. I am not very familiar with `nextjs / next-intl`. I hope someone can tell me where I should upload it. Thank you.

**Reply by amannn** [reply]
`public` should work (see [docs](https://nextjs.org/docs/app/building-your-application/optimizing/static-assets)), but maybe you have to put the file there before you build your app.

**Reply by aboutjquery** [reply]
> `public` should work (see [docs](https://nextjs.org/docs/app/building-your-application/optimizing/static-assets)), but maybe you have to put the file there before you build your app.

Work very well, thank you so much.
